### ISSUE TITLE
A11y keyboard focus control in the Dialog Component

### ISSUE THREAD

**Issue by DaveBest99** [angular/components#3769]
#### Bug, feature request, or proposal:
Bug: Dialog missing page structure.

#### What is the expected behavior?
The Dialog modal should define the main content region for context understanding.

#### What is the current behavior?
Currently the first focusable element in the Dialog is at the bottom of the page, skipping the screen reader user over all of the Dialog content.

#### What are the steps to reproduce?
See below.

Providing a Plunker (or similar) is the *best* way to get the team to see your issue.
Plunker template: https://goo.gl/DlHd6U

#### What is the use-case or motivation for changing an existing behavior?
Create a good screen reader user experience.

#### Which versions of Angular, Material, OS, browsers are affected?
Using screen reader JAWS17 with browsers Firefox, Internet Explorer 11, and Google Chrome, on Windows10.

#### Is there anything else we should know?
Screen reader user experience follows:

1. Press Tab key to the "Dailog with a result" demo component in the Tab Panel.
JAWS says "LAUNCH DIALOG button", and the braille display shows "LAUNCH DIALOG btn".
- Note, the typing error in "dailog".
- The CSS should be used to style text instead of using all capital letters in text labels.

2. Press Spacebar or Enter key on the "LAUNCH DIALOG" button.
JAWS says "option 1 button", and the braille display shows "OPTION 1 btn".
- This is a simple Dialog, and presents no accessibility challenge. However, the first focusable element is at the bottom of the Dialog modal, skipping over all the content.
- The md-dialog-title module is defined by the h1 tag, but the md-dialog-content module (Primary scrollable content of the dialog) and the 
md-dialog-actions module (Container for action buttons at the bottom of the dialog) have no defined structure markup.
- The CSS should be used to style text instead of using all capital letters "OPTION".
- Note, the buttons are labelled "OPTION 1" and "OPTION 2", which perform the same function "Close". The md-dialog-close is not clearly labelled.
- The Arrow and Tab keys function as expected in the Dialog modal, and the Option buttons close the Dialog and return the focus to the "LAUNCH DIALOG" element as expected. Pressing Down Arrow announces the message "You chose: Option 1" .

3. The "Overview" Tab of the Dialog demo has a "OPEN DIALOG" button, which does not open a Dialog modal. Pressing Spacebar or Enter key on this button appears to do nothing, as the screen reader does not change focus or announce any change. However, the text "Hi, I'm a dialog!" appears several times at the bottom of the screen. This may be a demo issue.

**Comment by vipdorian** [comment]
Hi DaveBeast99,
1. Please confirm: should pressing "Escape" key close the dialog without taking any action? 
See this other implementation which close the dialog on Escape : 
- http://demos.wijmo.com/5/Angular2/Explorer/Explorer/#/input/popup
- http://shlomiassaf.github.io/angular2-modal/#/js-native-demo

**Comment by DaveBest99** [comment]
Typically, the Escape key is considered to be a Cancel key. So, if a dialog has been modified, the Escape key should cancel the changes. However, there is no fixed rule for using the Escape key, but consistency, in all pages and components, is important to achieve a good user experience. That is, a ComboBox selection should perform the same result. A Dialog box may have an Ok and Cancel button, and the Escape key would mimic the Cancel Button. So, I would say Yes to your question.

David

**Comment by jelbourn** [comment]
The issues here all seem to be problems with the demos rather than the underlying dialog component. 
* The overview demo has no focusable elements inside of the dialog, so focus doesn't move _into_ the dialog. This isn't an intended use case for the component and the demo should be fixed.
* It seems like the demo under examples should have a paragraph tag inside of the `md-dialog-content` container.
